conans/client/tools/scm.py

```diff
@@ -62,14 +62,67 @@ def run(self, command):
                         return self._runner(command)
 
     def get_url_with_credentials(self, url):
-        if not self._username or not self._password:
+        if not self._username and not self._password:
             return url
-        if urlparse(url).password:
+        url_user = None
+        url_pass = None
+        scp_regex = re.compile("^([a-zA-Z0-9_]+)@([a-zA-Z0-9._-]+):(.*)$")
+        url_user_pass_regex = re.compile("^(file|http|git)s?:\/\/(\S+):(\S+)@("".*)$")
```

Comment:
What are the quotes in this group `("".*)` intended for?

Reply:
looks like a leftover